scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -619,163 +732,214 @@ class ProcessMonitor(ScalyrMonitor):
+        self.__target_pids = self._config.get('pid', default=None, convert_to=str)
 
-        if self.__commandline_matcher is None and self.__target_pid is None:
-            raise BadMonitorConfiguration('At least one of the following fields must be provide: commandline or pid',
-                                          'commandline')
+        # convert target pid into a list. Target pids can be a single pid or a CSV of pids
+        self.__target_pids = [int(x.strip()) for x in self.__target_pids.split(',')] if self.__target_pids else []
+        self.__target_pids = set(self.__target_pids)
+
+        # history of all metrics
+        self.metrics_history = defaultdict(dict)
+        # running total of metric values
+        self.running_total_metrics = {}
+
+        if not (self.__commandline_matcher or self.__target_pids):
+            raise BadMonitorConfiguration(
+                'At least one of the following fields must be provide: commandline or pid',
+                'commandline'
+            )
 
         # Make sure to set our configuration so that the proper parser is used.
         self.log_config = {
             'parser': 'agent-metrics',
             'path': 'linux_process_metrics.log',
         }
 
-    def __set_process(self, pid):
-        """Sets the id of the process for which this monitor instance should record metrics.
+    def record_metrics(self, pid, metrics):
+        """
+        For a process, record the metrics in a historical metrics collector
+        Collects the historical result of each metric per process in __metrics_history
+        which has form:
+
+        {
+          '<process id>: {
+                        <metric name>: [<metric at time 0>, <metric at time 1>.... ],
+                        <metric name>: [<metric at time 0>, <metric at time 1>.... ],
+                        }
+
+          '<process id>: {
+                        <metric name>: [<metric at time 0>, <metric at time 1>.... ],
+                        <metric name>: [<metric at time 0>, <metric at time 1>.... ],
+                        }
+        }
 
-        @param pid: The process id or None if there is no process to monitor.
-        @type pid: int or None
+        @param pid: Process ID
+        @param metrics: Collected metrics of the process
+        @type pid: int
+        @type metrics: dict
+        :return: None
+        """
+        for _metric, _metric_value in metrics.items():
+            if not self.metrics_history[pid].get(_metric):
+                self.metrics_history[pid][_metric] = []
+            self.metrics_history[pid][_metric].append(_metric_value)
+            # only keep the last 2 running history for any metric
+            self.metrics_history[pid][_metric] = self.metrics_history[pid][_metric][-2:]
+
+    def _reset_absolute_metrics(self):
+        """
+        At the beginning of each process metric calculation, the absolute metrics
```

Comment:
Let's make it clear here what you mean by absolute... So, say `.. each process metric calculation, the absolute (non-cumulative) metrics ...`